chaincode/src/dex/userpositions.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Gala Games Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ChainKey, ChainObject, PositionsObject } from "@gala-chain/api";
+import BigNumber from "bignumber.js";
+import { Exclude } from "class-transformer";
+import { IsString } from "class-validator";
+import { JSONSchema } from "class-validator-jsonschema";
+
+JSONSchema({
+  description: "Stores all the positions that the user has made in all the uniswap v3 pools."
+})
+export class UserPosition extends ChainObject {
+  @Exclude()
+  static INDEX_KEY = "POSITIONS";
```

Comment:
Should follow naming convention GC<acronym>
```suggestion
  static INDEX_KEY = "GCUPP"; -> GalaChain User Pool Positions
```